python/cuml/test/test_fil.py

```diff
@@ -486,16 +491,58 @@ def test_output_args(small_classifier_and_preds):
     assert array_equal(fil_preds, xgb_preds, 1e-3)
 
 
+def to_categorical(features, n_categorical):
+    # the main bottleneck (>80%) of to_categorical() is the pandas operations
+    n_features = features.shape[1]
+    df_cols = {}
+    # all categorical columns
+    cat_cols = features[:, :n_categorical]
+    cat_cols = cat_cols - cat_cols.min(axis=1, keepdims=True)  # range [0, ?]
+    cat_cols /= cat_cols.max(axis=1, keepdims=True)  # range [0, 1]
+    # round into rough_n_categories bins
+    cat_cols = (cat_cols * 100).astype(int)
+    for icol in range(n_categorical):
+        col = cat_cols[:, icol]
+        df_cols[icol] = pd.Series(pd.Categorical(col,
+                                                 categories=np.unique(col)))
+    # all numerical columns
+    for icol in range(n_categorical, n_features):
+        df_cols[icol] = pd.Series(features[:, icol])
+    # shuffle the columns around
+    seed(42)
```

Comment:
Please avoid setting any global state, and pass in `random_state` as a parameter.